root/static/scripts/common/utility/entityHref.js

```diff
@@ -87,7 +88,7 @@ function entityHref(
       break;
 
     default:
-      if (entityProps.mbid && entity.gid) {
+      if (entityProps.mbid && nonEmpty(entity.gid)) {
```

Comment:
It would always see it as non-empty, yes. But the `nonEmpty` call was mainly added to work around a sketchy-null error from Flow. If the unwrapped `gid` is empty, `entityHref` will return a nonsense result either way (because `id` will just remain an empty string).

Reply:
I'm guessing that's why I originally added https://github.com/metabrainz/musicbrainz-server/pull/3113#discussion_r1408145768 though.